helpers/search_processing_kendra_quickstart.sh

```diff
@@ -0,0 +1,419 @@
+        echo "Unable to load credentials from default profile. No credentials will be passed to the OpenSearch keystore."
+        echo "OpenSearch will use the default credential provider chain to access Kendra, which may rely on EC2 instance"
+        echo "profile credentials or credentials from ECS metadata service."
+      elif [ -z "${AWS_SESSION_TOKEN}" ]; then
+        echo "Using AWS credentials (aws_access_key_id and aws_secret_access_key) from default profile."
+      else 
+        echo "Using AWS credentials (aws_access_key_id, aws_secret_access_key, and aws_session_token) from default profile."
+      fi
+    else
+      >&2 echo "Environment variable AWS_SECRET_ACCESS_KEY is specified, but AWS_ACCESS_KEY_ID is not."
+      >&2 echo "Unable to determine which credentials to use."
+      exit 1
+    fi
+  else
+    # AWS_ACCCESS_KEY_ID is set
+    if [ -z "${AWS_SECRET_ACCESS_KEY:-}" ]; then
+      >&2 echo "Environment variable AWS_ACCESS_KEY_ID is specified, but AWS_SECRET_ACCESS_KEY is not."
+      >&2 echo "Unable to determine which credentials to use."
+      exit 1
+    else
+      if [ -n "${AWS_SESSION_TOKEN:-}"]; then
+        echo "Using credentials from environment (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, and AWS_SESSION_TOKEN)."
+      else
+        echo "Using credentials from environment (AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY)."
+      fi
+    fi
+  fi
+  if [ -z "${AWS_REGION:-}" ]; then
+    AWS_REGION=$(aws configure get region || echo)
+    if [ -n "${AWS_REGION:-}" ]; then
+      echo "Using AWS region ${AWS_REGION} from default profile"
+    else
+      >&2 echo "Argument [-r | --region] not specified and unable to infer region from default profile"
+      exit 1
+    fi
+  fi
+fi
+
+#
+# Infer Kendra endpoint from region if missing.
+#
+if [ -z "${KENDRA_RANKING_ENDPOINT:-}" ]; then
+  KENDRA_RANKING_ENDPOINT="https://kendra-ranking.${AWS_REGION}.api.aws"
+  echo "Missing argument [-e | --kendra-ranking-endpoint] -- using ${KENDRA_RANKING_ENDPOINT}"
+fi
+
+#
+# Create a new execution plan (with 1 rescore capacity unit) if explicitly requested.
+#
+if [ -n "${CREATE_EXECUTION_PLAN:-}" ]; then
+  if ! aws kendra-ranking help > /dev/null 2>1; then 
+    >&2 echo "AWS CLI does not support kendra-ranking service. Please install the latest AWS CLI."
+    exit 1
+  fi
+  if [ -n "${AWS_ACCESS_KEY_ID:-}" ] && [ -n "${AWS_SECRET_ACCESS_KEY:-}" ]; then
+    export AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY
+    if [ -n "${AWS_SESSION_TOKEN:-}" ]; then
+      export AWS_SESSION_TOKEN
+    fi
+  fi
+  EXECUTION_PLAN_ID=$(aws ${AWS_CLIENT_OPTS:-} --region ${AWS_REGION} --endpoint-url ${KENDRA_RANKING_ENDPOINT} \
+    kendra-ranking create-rescore-execution-plan --name "TestPlan" \
+      --description "Plan created by search_processing_kendra_quickstart.sh" \
+    --capacity-units RescoreCapacityUnits=1 --output text --query Id)
+  echo "Created execution plan with ID ${EXECUTION_PLAN_ID}. "
```

Comment:
The CLI command above has problems @msfroh 
1.   `--capacity-units RescoreCapacityUnits=1` results in error 
```
Parameter validation failed:
Missing required parameter in CapacityUnits: "QueryCapacityUnits"
Unknown parameter in CapacityUnits: "RescoreCapacityUnits", must be one of: QueryCapacityUnits
```
It's unnecessary.. please just remove it, since the default is '1' anyway.

2. When using a custom endpoint, can you also add `--no-verify-ssl` to prevent SSL failure of the command line.. both here, and in the decsribe-rescore-execution-plan loop below

3. Hardcoded name `--name "TestPlan"` causes problem when script is run a second time, and execution plan exists.  
```
An error occurred (ConflictException) when calling the CreateRescoreExecutionPlan operation: The same request is currently associated with a different client token. To create another execution plan, please change a request parameter and try your request again.
```
To address this please modify logic to catch this error, and proceed to reuse the existing execution plan.

Reply:
>Parameter validation failed:
Missing required parameter in CapacityUnits: "QueryCapacityUnits"
Unknown parameter in CapacityUnits: "RescoreCapacityUnits", must be one of: QueryCapacityUnits

The RescoreCapacityUnits parameter was renamed to QueryCapacityUnits a few days ago. I'll remove the argument if it defaults to 1, though.

> When using a custom endpoint, can you also add --no-verify-ssl to prevent SSL failure of the command line.. both here, and in the decsribe-rescore-execution-plan loop below

That's what the `${AWS_CLIENT_OPTS:-}` variable is for -- I'm running the command with `AWS_CLIENT_OPTS=--no-verify-ssl ./search_processing_kendra_quickstart.sh [...]`. As far as I know that should never be needed by real users, though. This is only because we're using pre-release endpoints with invalid certificates.

> Hardcoded name --name "TestPlan" causes problem when script is run a second time, and execution plan exists.

I think I can check if a plan with that name exists before trying to create it. If it's in a CREATING state, it can wait. If it's in ACTIVE state we can use it. If it has any other state, I think I would error out.

Reply:
1. Removed the `--capacity-units` param.
2. Use `AWS_CLIENT_OPTS=--no-verify-ssl`. (Not documenting this because AFAIK it should never be an issue for real AWS users and we don't want real AWS users disabling SSL verification.)
3. Added check for existing rescore execution plans named "TestPlan". (You can have multiple plans with the same name as long as some other parameter has changed, like the description or capacity units.)
    1. It will reuse the first plan with status "CREATING" or "ACTIVE".
    2. It ignores any plan with the name "DELETING" (since that doesn't prevent creating a new plan with those parameters).
    3. It fails if it finds a plan named "TestPlan" and some other status (since I don't know what that means).

Between this and the cleanup script, I'm finding this whole thing much friendlier. Let me know how it works for you.

Reply:
All sounds great.. I'll give it another whirl today. Thanks!